### PR TITLE
fix: add flexibility to classes

### PR DESCRIPTION
**Pull request by azayz**

### CODE REVIEW

File: clip_image_classifier.py [hunk 1]
```diff
@@ -47,23 +48,28 @@ def __init__(
 
     @requests
     def classify(self, docs: DocumentArray, parameters: dict, **kwargs):
-
+        classes = parameters.get('classes',self.classes)
+        if classes == None :
```
Comment (tadejsv) [inline clip_image_classifier.py:52]:
```suggestion
        if not classes :
```

File: clip_image_classifier.py [hunk 2]
```diff
@@ -47,23 +48,28 @@ def __init__(
...
+                '``classes`` parameter is mandatory. Pass it in parameters'
+                'or in constructor. e.g. ["This is a cat","This is a person", "This is a dog"]'
+            )
         for docs_batch in docs.traverse_flat(parameters.get('traversal_paths', self.traversal_paths)).batch(
             batch_size=parameters.get('batch_size', self.batch_size)
         ):
             image_batch = docs_batch.get_attributes('blob') 
             with torch.inference_mode():
-                input = self._generate_input_features(image_batch)
+                input = self._generate_input_features(classes,image_batch)
```
Comment (tadejsv) [inline clip_image_classifier.py:62]: So there should be a space after the comma - please format everything using `black`. Check out the pre-commit hooks from some other executor repos

File: tests/unit/test_clip_image_classifier.py [hunk 3]
```diff
@@ -12,6 +11,15 @@ def test_config():
     ex = Executor.load_config(str(Path(__file__).parents[2] / 'config.yml'))
     assert len(ex.classes) != 0
 
+def test_no_classes():
+    with pytest.raises(ValueError,match='``classes`` parameter is mandatory. Pass it in parameters'):
```
Comment (tadejsv) [inline tests/unit/test_clip_image_classifier.py:15]: Put only the statement that should generate the error in this context manager

File: clip_image_classifier.py [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 import torch
+from jina.logging.logger import JinaLogger
 
 class CLIPImageClassifier(Executor):
     """Classifies an image given a list of text classes using CLIP model"""
 
     def __init__(
         self,
-        classes: List[str] = ["this is a person","this is a cat","this is a dog"],
+        classes: Optional[List[str]] = None,
```
Comment (tadejsv) [inline clip_image_classifier.py:12]: I think we generally have some defaults in our executors, to make sure things work out of the box.

**Review by tadejsv — DISMISSED** [review]

File: clip_image_classifier.py (hunk 4 above)
Comment (makram93) [reply clip_image_classifier.py:12]: Can this [vocab file](https://huggingface.co/openai/clip-vit-base-patch32/blob/main/vocab.json) be used as to fill the default `classes` args? @azayz

File: clip_image_classifier.py [hunk 5]
```diff
@@ -47,23 +48,28 @@ def __init__(
 
     @requests
     def classify(self, docs: DocumentArray, parameters: dict, **kwargs):
-
+        classes = parameters.get('classes',self.classes)
```
Comment (makram93) [inline clip_image_classifier.py:51]:
```suggestion
        classes = parameters.get('classes', self.classes)
```
Formatting

**Review by makram93 — CHANGES_REQUESTED** [review]
Minor comment: format the file and see if we can make use of this [vocab file](https://huggingface.co/openai/clip-vit-base-patch32/blob/main/vocab.json) to fill the default `classes` param, so that it can work out of the box.

File: clip_image_classifier.py (hunk 4 above)
Comment (azayz) [reply clip_image_classifier.py:12]: It's a good idea but these are words used for tokenization, to my knowledge classes should be phrases describing what the image is about

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**